jcache/commons/src/main/java/org/infinispan/jcache/AbstractJCacheNotifier.java

```diff
@@ -37,6 +42,13 @@
 
    private static final boolean isTrace = log.isTraceEnabled();
 
+   private static final ConcurrentLinkedQueue<CountDownLatch> EMPTY_QUEUE = new ConcurrentLinkedQueue<CountDownLatch>() {
```

Comment:
I don't understand this. Will you ever modify this? If not, why is a ConcurrentLinkedQueue? And if you're going to modify it, won't you need a different equals implementation?